app/models/post.rb

```diff
@@ -217,6 +222,10 @@ def limit_posts_per_day
     end
   end
 
+  def badges_granted
+    user_badges.select { |user_badge| user_badge.user_id == user_id }
```

Comment:
```suggestion
    user_badges.where("user_badges.user_id = ?", user_id)
```

I believe we can do the filtering directly in the database to avoid over fetching from the database.